consoleme/lib/slack.py

```diff
@@ -0,0 +1,109 @@
+import sys
+
+import requests
+import tornado.escape
+
+from consoleme.config import config
+from consoleme.lib.plugins import get_plugin_by_name
+from consoleme.lib.policies import get_policy_request_uri_v2
+from consoleme.models import ExtendedRequestModel
+
+log = config.get_logger()
+stats = get_plugin_by_name(config.get("plugins.metrics", "default_metrics"))()
+
+
+async def send_slack_notification_new_request(
+    extended_request: ExtendedRequestModel, admin_approved, approval_probe_approved
+):
+    """
+    Sends a notification using specified webhook URL about a new request created
+    """
+    if not config.get("slack.notifications_enabled", False):
+        return
+
+    function = f"{__name__}.{sys._getframe().f_code.co_name}"
+    requester = extended_request.requester_email
+    arn = extended_request.principal.principal_arn
+    stats.count(function, tags={"user": requester, "arn": arn})
+
+    log_data: dict = {
+        "function": f"{__name__}.{sys._getframe().f_code.co_name}",
+        "user": requester,
+        "arn": arn,
+        "message": "Incoming request for slack notification",
+        "request": extended_request.dict(),
+        "admin_approved": admin_approved,
+        "approval_probe_approved": approval_probe_approved,
+    }
+    log.debug(log_data)
+    slack_webhook_url = config.get("slack.webhook_url")
+    if not slack_webhook_url:
+        log_data["message"] = "Missing webhook URL for slack notification"
+        log.error(log_data)
+        return
+    payload = await get_payload(
+        extended_request, requester, arn, admin_approved, approval_probe_approved
+    )
+    resp = requests.post(slack_webhook_url, json=payload)
```

Comment:
Since requests is blocking, can you switch it out for AsyncHttpClient? There's some example usage lying around, IE consoleme/lib/account_indexers/swag.py

Reply:
Yup, updated!